packages/flutter/lib/src/widgets/router.dart

```diff
@@ -1094,7 +1094,7 @@ abstract class RouteInformationParser<T> {
   /// Consider using a [SynchronousFuture] if the result can be computed
   /// synchronously, so that the [Router] does not need to wait for the next
   /// microtask to pass the data to the [RouterDelegate].
-  Future<T> parseRouteInformation(RouteInformation routeInformation);
```

Comment:
I have thought about how to not break the customer by creating another API which delegate the the original API by default. However, that will then require developer to override both APIs if they want to use the new API (because we can't create a default implementation for the old API)

Reply:
Why can we not provide an (empty) default implementation for the old API? (Instead of empty we may actually want to throw.). If we do that, you could add a new `betterParseRouteInformation` (needs a better name) with the additional argument, that by default just calls through to the old API. Now you just need to override one of them.

(It will still break people, who or implementing this class instead of extending it, not sure how common that is...)

Reply:
Is there a way to provide the default implementation for a method that return Future<T>? I remember if you do Futrue<T>.value(null), it may crash if T is not null type

Reply:
This should work as default implementation, no?

```dart
throw UnimplementedError();
```